pandas/core/indexes/interval.py

```diff
@@ -787,6 +788,50 @@ def _format_data(self, name=None) -> str:
         # name argument is unused here; just for compat with base / categorical
         return self._data._format_data() + "," + self._format_space()
 
+    # --------------------------------------------------------------------
+    # Set Operations
+
+    def _intersection(self, other, sort):
```

Comment:
you could probably do this more generally (e.g. push it down into Index itself). but of course for later.